### PR TITLE
[Keyboard] Add pimentoso/paddino02

### PR DESCRIPTION
**Pull request by Pimentoso**
Added QMK support for my macropad, instead of relying to kbfirmware.com json files.

Project page: https://github.com/Pimentoso/Paddino02

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/pimentoso/paddino02/rev1/config.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xD00D
+#define PRODUCT_ID      0x0020
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    jlcpcb
```
Comment (Rionlion100) [inline keyboards/pimentoso/paddino02/rev1/config.h:9]: This should not be JLCPCB but instead your username/brand

File: keyboards/pimentoso/paddino02/rev1/rev1.c [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/* Copyright 2017 REPLACE_WITH_YOUR_NAME
```
Comment (Rionlion100) [inline keyboards/pimentoso/paddino02/rev1/rev1.c:1]:
```suggestion
/* Copyright 2017 Mattia Dal Ben <matthewdibi@gmail.com>
```

File: keyboards/pimentoso/paddino02/rev2/left/config.h [hunk 3]
```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xD00D
+#define PRODUCT_ID      0x0021
+#define DEVICE_VER      0x0002
+#define MANUFACTURER    jlcpcb
```
Comment (Rionlion100) [inline keyboards/pimentoso/paddino02/rev2/left/config.h:9]: This should not be JLCPCB but instead your username/brand

File: keyboards/pimentoso/paddino02/rev2/left/left.c [hunk 4]
```diff
@@ -0,0 +1,16 @@
+/* Copyright 2017 REPLACE_WITH_YOUR_NAME
```
Comment (Rionlion100) [inline keyboards/pimentoso/paddino02/rev2/left/left.c:1]:
```suggestion
/* Copyright 2017 Mattia Dal Ben <matthewdibi@gmail.com>
```

File: keyboards/pimentoso/paddino02/rev2/right/right.c [hunk 5]
```diff
@@ -0,0 +1,16 @@
+/* Copyright 2017 REPLACE_WITH_YOUR_NAME
```
Comment (Rionlion100) [inline keyboards/pimentoso/paddino02/rev2/right/right.c:1]:
```suggestion
/* Copyright 2017 Mattia Dal Ben <matthewdibi@gmail.com>
```

File: keyboards/pimentoso/paddino02/rev2/right/config.h [hunk 6]
```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xD00D
+#define PRODUCT_ID      0x0022
+#define DEVICE_VER      0x0002
+#define MANUFACTURER    jlcpcb
```
Comment (Rionlion100) [inline keyboards/pimentoso/paddino02/rev2/right/config.h:9]: This should not be JLCPCB but instead your username/brand

File: keyboards/pimentoso/paddino02/rules.mk [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+MOUSEKEY_ENABLE ?= yes # Mouse keys
+EXTRAKEY_ENABLE ?= yes # Audio control and System control
+CONSOLE_ENABLE ?= no # Console for debug
+COMMAND_ENABLE ?= no # Commands for debug and configuration
+SLEEP_LED_ENABLE ?= no # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= yes # USB Nkey Rollover
+BACKLIGHT_ENABLE ?= no # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+RGBLIGHT_ENABLE ?= no
+KEY_LOCK_ENABLE ?= no
```
Comment (ridingqwerty) [inline keyboards/pimentoso/paddino02/rules.mk:58]:
```suggestion
BOOTMAGIC_ENABLE = no   # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes   # Mouse keys
EXTRAKEY_ENABLE = yes   # Audio control and System control
CONSOLE_ENABLE = no     # Console for debug
COMMAND_ENABLE = no     # Commands for debug and configuration
SLEEP_LED_ENABLE = no   # Breathing sleep LED during USB suspend
NKRO_ENABLE = yes       # USB Nkey Rollover
BACKLIGHT_ENABLE = no   # Enable keyboard backlight functionality
AUDIO_ENABLE = no
RGBLIGHT_ENABLE = no
KEY_LOCK_ENABLE = no
```
The `?=` shouldn't be needed as this is a top-level Makefile for this board.

File: keyboards/pimentoso/paddino02/rev1/rev1.h [hunk 8]
```diff
@@ -0,0 +1,16 @@
+#ifndef KB_H
+#define KB_H
+
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev1/rev1.h:3]:
```suggestion
#pragma once

```

File: keyboards/pimentoso/paddino02/rev1/rev1.h [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+#include "quantum.h"
+
+#define KEYMAP( \
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev1/rev1.h:6]:
```suggestion
#define LAYOUT( \
```

File: keyboards/pimentoso/paddino02/rev1/rev1.h [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+	K00, K01, K02, K03, \
+	K10, K11, K12, K13, \
+	          K22, K23  \
+) { \
+	{ K00,   K01,   K02,   K03 }, \
+	{ K10,   K11,   K12,   K13 }, \
+	{ KC_NO, KC_NO, K22,   K23 }  \
+}
+
+#endif
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev1/rev1.h:16]:
```suggestion
```

File: keyboards/pimentoso/paddino02/rev1/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+
+void matrix_scan_user(void) {
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+	return true;
+}
+
+void led_set_user(uint8_t usb_led) {
+}
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev1/keymaps/default/keymap.c:32]:
```suggestion
```

File: keyboards/pimentoso/paddino02/rev1/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,32 @@
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	KEYMAP(
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev1/keymaps/default/keymap.c:5]:
```suggestion
	LAYOUT(
```

File: keyboards/pimentoso/paddino02/rev1/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+		KC_ESC, KC_BTN1, KC_UP, KC_BTN2,
+		KC_LSFT, KC_LEFT, KC_DOWN, KC_RGHT,
+		MO(1), KC_SPC),
+
+	KEYMAP(
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev1/keymaps/default/keymap.c:10]:
```suggestion
	LAYOUT(
```

File: keyboards/pimentoso/paddino02/rev2/left/config.h [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
+
+#define RGBLED_NUM 0
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#define RGBLIGHT_SLEEP
+#define RGBLIGHT_LIMIT_VAL undefined
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev2/left/config.h:48]:
```suggestion
```

File: keyboards/pimentoso/paddino02/rev2/left/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,32 @@
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	KEYMAP(
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev2/left/keymaps/default/keymap.c:5]:
```suggestion
	LAYOUT(
```

File: keyboards/pimentoso/paddino02/rev2/left/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+		KC_ESC, KC_BTN1, KC_UP, KC_BTN2,
+		KC_LSFT, KC_LEFT, KC_DOWN, KC_RGHT,
+		MO(1), KC_SPC),
+
+	KEYMAP(
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev2/left/keymaps/default/keymap.c:10]:
```suggestion
	LAYOUT(
```

File: keyboards/pimentoso/paddino02/rev2/left/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+
+void matrix_scan_user(void) {
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+	return true;
+}
+
+void led_set_user(uint8_t usb_led) {
+}
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev2/left/keymaps/default/keymap.c:32]:
```suggestion
```

File: keyboards/pimentoso/paddino02/rev2/left/left.h [hunk 18]
```diff
@@ -0,0 +1,16 @@
+#ifndef KB_H
+#define KB_H
+
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev2/left/left.h:3]:
```suggestion
#pragma once
```

File: keyboards/pimentoso/paddino02/rev2/left/left.h [hunk 19]
```diff
@@ -0,0 +1,16 @@
...
+#include "quantum.h"
+
+#define KEYMAP( \
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev2/left/left.h:6]:
```suggestion
#define LAYOUT( \
```

File: keyboards/pimentoso/paddino02/rev2/left/left.h [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+	K00, K01, K02, K03, \
+	K10, K11, K12, K13, \
+	          K22, K23  \
+) { \
+	{ K00,   K01,   K02,   K03 }, \
+	{ K10,   K11,   K12,   K13 }, \
+	{ KC_NO, KC_NO, K22,   K23 }  \
+}
+
+#endif
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev2/left/left.h:16]:
```suggestion
```

File: keyboards/pimentoso/paddino02/rev2/right/keymaps/default/keymap.c [hunk 21]
```diff
@@ -0,0 +1,32 @@
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	KEYMAP(
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev2/right/keymaps/default/keymap.c:5]:
```suggestion
	LAYOUT(
```

File: keyboards/pimentoso/paddino02/rev2/right/keymaps/default/keymap.c [hunk 22]
```diff
@@ -0,0 +1,32 @@
...
+		KC_ESC, KC_BTN2, KC_UP, KC_BTN1,
+		KC_LSFT, KC_RGHT, KC_DOWN, KC_LEFT,
+		MO(1), KC_SPC),
+
+	KEYMAP(
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev2/right/keymaps/default/keymap.c:10]:
```suggestion
	LAYOUT(
```

File: keyboards/pimentoso/paddino02/rev2/right/keymaps/default/keymap.c [hunk 23]
```diff
@@ -0,0 +1,32 @@
...
+
+void matrix_scan_user(void) {
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+	return true;
+}
+
+void led_set_user(uint8_t usb_led) {
+}
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev2/right/keymaps/default/keymap.c:32]:
```suggestion
```

File: keyboards/pimentoso/paddino02/rev2/right/right.h [hunk 24]
```diff
@@ -0,0 +1,16 @@
+#ifndef KB_H
+#define KB_H
+
+#include "quantum.h"
+
+#define KEYMAP( \
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev2/right/right.h:6]:
```suggestion
#define LAYOUT( \
```

File: keyboards/pimentoso/paddino02/rev2/right/right.h [hunk 25]
```diff
@@ -0,0 +1,16 @@
+#ifndef KB_H
+#define KB_H
+
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev2/right/right.h:3]:
```suggestion
#pragma once

```

File: keyboards/pimentoso/paddino02/rev2/right/right.h [hunk 26]
```diff
@@ -0,0 +1,16 @@
...
+	K00, K01, K02, K03, \
+	K10, K11, K12, K13, \
+	K20, K21  \
+) { \
+	{ K03,   K02,   K01,   K00 }, \
+	{ K13,   K12,   K11,   K10 }, \
+	{ K21,   K20,   KC_NO, KC_NO }  \
+}
+
+#endif
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev2/right/right.h:16]:
```suggestion
```

File: keyboards/pimentoso/paddino02/rules.mk (hunk 7 above)
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rules.mk:58]:
```suggestion
# Bootloader selection
#   Teensy       halfkay
#   Pro Micro    caterina
#   Atmel DFU    atmel-dfu
#   LUFA DFU     lufa-dfu
#   QMK DFU      qmk-dfu
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```

**Comment by Pimentoso** [comment]
Applied the requested changes. Thanks guys.

File: keyboards/pimentoso/paddino02/rev1/config.h [hunk 27]
```diff
@@ -0,0 +1,51 @@
...
+
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
+
+#define RGBLED_NUM 0
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#define RGBLIGHT_SLEEP
+#define RGBLIGHT_LIMIT_VAL undefined
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev1/config.h:50]:
```suggestion
```

File: keyboards/pimentoso/paddino02/readme.md [hunk 28]
```diff
@@ -0,0 +1,29 @@
...
+
+`make pimentoso/paddino02/rev2/right:default`
+`make pimentoso/paddino02/rev2/left:default`
+`make pimentoso/paddino02/rev1:default` (legacy version of the PCB)
+
+### To flash:
+
+`make pimentoso/paddino02/rev2/right:default:avrdude`
+`make pimentoso/paddino02/rev2/left:default:avrdude`
+`make pimentoso/paddino02/rev1:default:avrdude` (legacy version of the PCB)
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/readme.md:23]:
```suggestion
`make pimentoso/paddino02/rev2/right:default:flash`
`make pimentoso/paddino02/rev2/left:default:flash`
`make pimentoso/paddino02/rev1:default:flash` (legacy version of the PCB)
```

File: keyboards/pimentoso/paddino02/rev2/right/config.h [hunk 29]
```diff
@@ -0,0 +1,49 @@
...
+
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
+
+#define RGBLED_NUM 0
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#define RGBLIGHT_SLEEP
+#define RGBLIGHT_LIMIT_VAL undefined
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/rev2/right/config.h:48]:
```suggestion
```

File: keyboards/pimentoso/paddino02/config.h [hunk 30]
```diff
@@ -0,0 +1,20 @@
+/*
+Copyright 2018 Mattia Dal Ben <matthewdibi@gmail.com>
```
Comment (zvecr) [inline keyboards/pimentoso/paddino02/config.h:2]:
```suggestion
Copyright 2020 Michele Ferri <zomgsako@gmail.com>
```

File: keyboards/pimentoso/paddino02/rev1/config.h [hunk 31]
```diff
@@ -0,0 +1,48 @@
...
+#define UNUSED_PINS
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION ROW2COL
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 3
+#endif
```
Comment (fauxpark) [inline keyboards/pimentoso/paddino02/rev1/config.h:29]:
```suggestion
#define BACKLIGHT_LEVELS 3
```

File: keyboards/pimentoso/paddino02/rev2/left/config.h [hunk 32]
```diff
@@ -0,0 +1,48 @@
...
+#define UNUSED_PINS
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 3
+#endif
```
Comment (fauxpark) [inline keyboards/pimentoso/paddino02/rev2/left/config.h:29]:
```suggestion
#define BACKLIGHT_LEVELS 3
```

File: keyboards/pimentoso/paddino02/rev2/right/config.h [hunk 33]
```diff
@@ -0,0 +1,48 @@
...
+#define UNUSED_PINS
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 3
+#endif
```
Comment (fauxpark) [inline keyboards/pimentoso/paddino02/rev2/right/config.h:29]:
```suggestion
#define BACKLIGHT_LEVELS 3
```

File: keyboards/pimentoso/paddino02/rules.mk [hunk 34]
```diff
@@ -0,0 +1,26 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   atmega32a    bootloadHID
```
Comment (fauxpark) [inline keyboards/pimentoso/paddino02/rules.mk:10]:
```suggestion
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
```

File: keyboards/pimentoso/paddino02/rules.mk [hunk 35]
```diff
@@ -0,0 +1,26 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
```
Comment (fauxpark) [inline keyboards/pimentoso/paddino02/rules.mk:10]:
```suggestion
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
```

**Comment by drashna** [comment]
Wait, why the left and right revisions?

**Comment by Pimentoso** [comment]
@drashna same pcb, you flip it over but the pro micro is always soldered on the udnerside, so the pins of the matrix change

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**